src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/UtilityExtension.kt

```diff
@@ -138,6 +149,19 @@ class UtilityExtension : Extension() {
             }
         }
 
+        event<ThreadUpdateEvent> {
+            action {
+                val channel = event.channel
+
+                if (channel.isArchived && threads.get(channel) != null && threads.get(channel)!!.save) {
```

Comment:
You have two calls to the database here - this should be one call stored in a variable and checked. It'd also save you from using `!!`.